lib/_http_client.js

```diff
@@ -569,10 +569,20 @@ ClientRequest.prototype.setTimeout = function(msecs, callback) {
 };
 
 ClientRequest.prototype.setNoDelay = function() {
-  this._deferToConnect('setNoDelay', arguments);
+  var argsLength = arguments.length;
+  var args = new Array(argsLength);
+  for (var i = 0; i < argsLength; i++) {
+    args[i] = arguments[i];
+  }
```

Comment:
IMHO I've always found that style (putting `if`/loop/etc. bodies on the same line) less readable when visually scanning code, but I would suggest removing the braces for one line bodies.